jsonschema-validation.xml

```diff
@@ -1015,15 +1015,17 @@
 
             <section title="contentSchema">
                 <t>
-                    If the instance is a string, and if "contentMediaType" is present, this
-                    property contains a schema which describes the structure of the string.
+                    If the instance is a string containing valid JSON, and if "contentMediaType" is
```

Comment:
There is no need for the instance to be JSON.  `contentMediaType` specifically either forbids or recommends against (I forget) attempting to parse the instance according to the media type.

Reply:
This is a point that's always confused me. If the instance isn't JSON, how can the contents of "contentSchema", a JSON schema, validate it?

Reply:
@ssilverman JSON Schema is not defined over JSON, it is defined over a data model that derives from (but is not precisely identical to) the JSON data model.  The Core spec has a whole section on the data model, and notes that JSON Schema can be used with any media type that can be mapped into that data model.  Defining such mappings is outside the scope of this specification.  A lot of them are trivial at least in typical usage: YAML, for example, as long as you don't use the advanced features that most people don't know exist.

So the instance media type may not have been JSON (or at least not plain `application/json`) in the first place, and the content in the string might be that same other media type.

JSON Schema is very careful not to put restrictions on the `content*` keywords regarding media types.  This is also why implementations must not, by default, try to parse the embedded content and apply the schema.  That's an open-ended burden, and parsing and loading an arbitrary media type is a major security concern.

Reply:
It sounds like the change at this position should be reverted entirely? is there anything else to discuss there?

Reply:
@karenetheridge I think

> this property contains a schema

is the language that @ssilverman originally found confusing.  Because it almost sounds like a syntactic requirement but is not.  Since he fixed the syntactic requirement further down, maybe there's a better wording that doesn't cause that confusion?  Or we can just revert it and assume that the last paragraph covers the formalities.

Reply:
How about:

> This keyword only applies to instance data of the "string" type. If "contentMediaType" is present, this property contains a schema which describes the structure of the string.

Reply:
We shouldn't revert this entirely, because it still needs to add the "MUST". I've force pushed that one change.